homeassistant/config_entries.py

```diff
@@ -2449,7 +2471,10 @@ def _async_in_progress(
     async def async_step_ignore(self, user_input: dict[str, Any]) -> ConfigFlowResult:
         """Ignore this config flow."""
         await self.async_set_unique_id(user_input["unique_id"], raise_on_progress=False)
-        return self.async_create_entry(title=user_input["title"], data={})
+        data = {}
+        if "discovery_key" in user_input:
+            data["discovery_keys"] = [user_input["discovery_key"]]
```

Comment:
I don't completely follow how storing a single key here plays with updating many known keys with a discovery key above when finishing the flow.

Reply:
There will only be a single active discovery flow per device, also when the integration has multiple discovery sources for the same device. `async_step_ignore` is called when the user ignores a discovered device or service, we then store the key for the flow being ignored.

Once the ignore config entry is created, the code above will make sure the discovery key is kept up to date since it may not be stable unlike the unique id.

Reply:
That makes sense. Please add it to the docstring. :+1: